airflow/providers/apache/hive/hooks/hive.py

```diff
@@ -170,6 +178,10 @@ def _prepare_cli_cmd(self) -> list[Any]:
                 if ";" in proxy_user:
                     raise RuntimeError("The proxy_user should not contain the ';' character")
                 jdbc_url += f";principal={template};{proxy_user}"
+                if self.high_availability:
+                    if proxy_user:
+                        jdbc_url += ";"
+                    jdbc_url += "serviceDiscoveryMode=zooKeeper;ssl=true;zooKeeperNamespace=hiveserver2"
```

Comment:
Fields needed in presence of HA, usually zookeeper will be installed in HA hive.